certified-connectors/Adobe Creative Cloud Libraries/apiDefinition.swagger.json

```diff
@@ -0,0 +1,616 @@
+                        }
+                      }
+                    }
+                  },
+                  "description": "data"
+                },
+                "config": {
+                  "type": "object",
+                  "properties": {
+                    "url": {
+                      "type": "string",
+                      "description": "url",
+                      "title": "Callback URL",
+                      "x-ms-visibility": "internal",
+                      "x-ms-notification-url": true
+                    }
+                  },
+                  "description": "config",
+                  "required": [
+                    "url"
+                  ]
+                }
+              },
+              "required": [
+                "config"
+              ]
+            }
+          }
+        ]
+      }
+    },
+    "/libraries": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "library_urn": {
+                    "type": "string",
+                    "description": "library_urn"
+                  },
+                  "name": {
+                    "type": "string",
+                    "description": "name"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Libraries List",
+        "operationId": "GetLibrariesList",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "$ref": "#/parameters/clientId_OAuthWeb"
+          },
+          {
+            "name": "selector",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "default": "all",
+            "x-ms-visibility": "internal"
+          }
+        ],
+        "description": "Get Libraries List."
+      }
+    },
+    "/webhooks/power-automate/{x_adobe_event_code}/unsubscribe/{id}": {
+      "delete": {
+        "responses": {
+          "200": {
+            "description": "OK"
+          }
+        },
+        "summary": "Delete action",
+        "description": "Delete action",
+        "operationId": "DeleteAction",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "name": "x_adobe_event_code",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-url-encoding": "single"
+          },
+          {
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-url-encoding": "single"
+          }
+        ]
+      }
+    }
+  },
+  "definitions": {},
+  "parameters": {
+    "clientId_OAuthWeb": {
+      "in": "header",
+      "name": "x-api-key",
+      "type": "string",
+      "default": "{{ client id }}",
```

Comment:
@dches Can you please let us know the reason to pass client id in the header of the actions?

Reply:
@sriyen-msft Adobe API requires this header for authorization.

Reply:
@dches It is not a good idea to have secure information in the swagger. Can you please start an email thread with us(ConnectorPartnerMgmtTeam@service.microsoft.com) to understand the risks to have a client id in the header of all requests that are defined in the swagger? And we/our engineering team may provide you if we have a resolution.

Reply:
@sriyen-msft Ok, I will talk about this header with my team.

Reply:
@sriyen-msft  We removed this parameter from the swagger file.